docs/host-powershell/README.md

```diff
@@ -0,0 +1,149 @@
+
+`PowerShellAssemblyLoadContextInitializer` exposes 2 APIs for this purpose:
+`SetPowerShellAssemblyLoadContext` and `InitializeAndCallEntryMethod`.
+They are for different scenarios:
+
+- For `SetPowerShellAssemblyLoadContext`,
+it's designed to be used by a native host whose TPA (Trusted Platform Assemblies) list doesn't include powershell assemblies,
+such as the in-box `powershell.exe` and other native CoreCLR host in Nano Server.
+When using this API, instead of setting up a new load context,
+`PowerShellAssemblyLoadContextInitializer` will register a handler to the [Resolving][] event of the default load context.
+Then PowerShell Core will depend on the default load context to handle TPA and the `Resolving` event to handle other assemblies.
+
+- For `InitializeAndCallEntryMethod`,
+it's designed to be used with `dotnet.exe` where the TPA list includes powershell assemblies.
+When using this API, `PowerShellAssemblyLoadContextInitializer` will set up a new load context to handle all assemblies.
+PowerShell Core itself also uses this API for [bootstrapping][].
+
+This documentation only covers the `InitializeAndCallEntryMethod` API,
+as it's what you need when building a .NET Core application with .NET CLI.
+
+### Example
+
+The following code is how you host Windows PowerShell in an application.  
+It's straight forward that you can directly run your business logic code from the application entry point `'Main'` method.
+
+```CSharp
+// MyApp.exe
+using System;
+using System.Management.Automation;
+
+public class Program
+{
+    static void Main(string[] args)
+    {
+        // My business logic code
+        using (PowerShell ps = PowerShell.Create())
+        {
+            var results = ps.AddScript("Get-Command Write-Output").Invoke();
+            Console.WriteLine(results[0].ToString());
+        }
+    }
+}
+```
+
+However, when it comes to hosting PowerShell Core, there will be a layer of redirection for the powershell load context to take effect.
+In a .NET Core application, the entry point assembly that contains the `'Main'` method is loaded in the default load context,
+and thus all assemblies referenced by the entry point assembly, implicitly or explicitly, will also be loaded into the default load context.
+
+In order to have the powershell load context to control assembly loading for the execution of an application,
+the business logic code needs to be extracted out of the entry point assembly and put into a different assembly, say `'Logic.dll'`.
+The entry point `'Main'` method shall do one thing only -- let the powershell load context load `'Logic.dll'` and start the execution of the business logic.
+Once the execution starts this way, all further assembly loading requests will be handled by the powershell load context.
+
+So the above example needs to be altered as follows in a .NET Core application:
```

Comment:
Same here. The code about hosting windows powershell is used as a comparison to show how this familiar code should be altered when it comes to .NET Core 1.1.